src/js/health-records/reducers/form.js

```diff
@@ -33,13 +34,19 @@ export default function form(state = initialState, action) {
       return set(`reportTypes.${action.reportType}`, action.checked, state);
     case 'ALL_REPORTS_TOGGLED':
       return set('reportTypes', mapValues(state.reportTypes, () => action.checked), state);
+    case 'FORM_SUBMITTING':
+      return set('inProgress', true, state);
     case 'FORM_SUCCESS':
       return set('ready', true, {
         ...state,
+        inProgress: false,
         requestDate: new Date().toISOString(),
       });
     case 'FORM_FAILURE':
-      return set('ready', false, state);
+      return set('ready', false, {
+        ...state,
+        inProgress: false,
+      });
```

Comment:
Although I didn't catch it in previous reviews, this and the change above both seem to me like a strange hybrid use of `set`. If we're using the rest operator anyway, the return value might as well be `{ ...state, ready: false, inProgress: false }`.